### PR TITLE
fix: preserve connector HTTP client error semantics

### PR DESCRIPTION
**Pull request by gopikrishna000**
## Description

- Map common connector HTTP error statuses to canonical gRPC codes, including `409 -> ABORTED`.
- Treat otherwise-unmapped connector 4xx responses as `INVALID_ARGUMENT` instead of `UNKNOWN`.
- Keep unclassified connector 5xx responses as `UNKNOWN` so they are not mislabeled as internal UCS failures.
- Preserve the exact upstream status in `ConnectorError.http_status_code`.
- Add one table-driven unit test for the mapping and fallback boundaries.

## Motivation and Context

PPRO can return HTTP 409 when an idempotency key is reused with a different request body. Previously, 409 fell through to gRPC `UNKNOWN`; UCS HTTP mode then translated that to HTTP 500 even though the structured connector details correctly contained `http_status_code: 409`.

This change maps generic conflicts to gRPC `ABORTED`, which the HTTP bridge returns as HTTP 409, and prevents other connector 4xx responses from being reported as UCS 500 errors.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

- `cargo fmt --all -- --check`
- `git diff --check`
- `cargo test -p ucs_env`
- `cargo clippy -p ucs_env --all-targets -- -D warnings`

### CODE REVIEW

File: crates/common/ucs_env/src/error.rs [hunk 1]
```diff
@@ -243,6 +236,28 @@ impl ToGrpcStatus for ConnectorError {
...
+        401 => tonic::Code::Unauthenticated,
+        403 => tonic::Code::PermissionDenied,
+        404 => tonic::Code::NotFound,
+        408 | 504 => tonic::Code::DeadlineExceeded,
+        409 => tonic::Code::Aborted,
+        412 => tonic::Code::FailedPrecondition,
+        429 => tonic::Code::ResourceExhausted,
+        501 => tonic::Code::Unimplemented,
+        502 | 503 => tonic::Code::Unavailable,
+        400..=499 => tonic::Code::InvalidArgument,
```
Comment (Aishwariyaa-Anand) [inline crates/common/ucs_env/src/error.rs]: Any connector 4xx not explicitly matched in connector_http_status_to_grpc_code (e.g. 400, 405, 410, 415, 422, 423, etc.) gets classified as InvalidArgument and surfaces to the UCS caller as HTTP 400 — regardless of what the connector's actual status was.

Comment (gopikrishna000) [reply crates/common/ucs_env/src/error.rs]: this is http status to grpc code conversion, there are no wider enums to support all possible http status code
But for response for ucs-http endpoint, if we need to extract connector's real status code, then we need to rely on existing ConnectorError.CONNECTOR_ERROR_RESPONSE.status_code

File: crates/common/ucs_env/src/error.rs [hunk 2]
```diff
@@ -243,6 +236,28 @@ impl ToGrpcStatus for ConnectorError {
...
+///
+/// The exact connector status is retained in `ConnectorError.http_status_code`.
+/// Unknown connector 4xx responses remain caller-visible errors rather than being
+/// converted to `Unknown`, which the HTTP bridge exposes as a UCS 500 response.
+fn connector_http_status_to_grpc_code(status_code: u16) -> tonic::Code {
+    match status_code {
+        401 => tonic::Code::Unauthenticated,
+        403 => tonic::Code::PermissionDenied,
+        404 => tonic::Code::NotFound,
+        408 | 504 => tonic::Code::DeadlineExceeded,
```
Comment (Aishwariyaa-Anand) [inline crates/common/ucs_env/src/error.rs]: DeadlineExceeded will be 504, should we include 408 also here?

**Review by Aishwariyaa-Anand — DISMISSED** [review]

File: crates/grpc-server/grpc-server/src/http/error.rs [hunk 3]
```diff
@@ -187,17 +187,23 @@ fn extract_error_details_from_status(status: &tonic::Status) -> Option<ErrorDeta
     None
 }
 
-/// Extract the connector's exact HTTP error status when the gRPC details identify a
+/// Extract the connector's exact HTTP error status when the decoded details identify a
 /// connector-originated 4xx/5xx response.
-fn connector_http_status_from_grpc_status(status: &tonic::Status) -> Option<StatusCode> {
-    let connector_error =
-        grpc_api_types::payments::ConnectorError::decode(status.details()).ok()?;
+fn connector_http_status_from_error_details(details: Option<&ErrorDetails>) -> Option<StatusCode> {
```
Comment (Aishwariyaa-Anand) [inline crates/grpc-server/grpc-server/src/http/error.rs:192]: do not use early returns

**Review by Aishwariyaa-Anand — APPROVED** [review]

**Review by AmitsinghTanwar007 — APPROVED** [review]
